tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_provision.c

```diff
@@ -172,6 +205,45 @@ static void test_provisioner_pb_adv_multi(void)
 	PASS();
 }
 
+static void test_provisioner_pb_adv_reprovision(void)
+{
+	int err;
+
+	k_sem_init(&prov_sem, 0, 1);
+
+	bt_mesh_device_setup();
+
+	err = bt_mesh_cdb_create(test_net_key);
+	ASSERT_OK(err, "Failed to create CDB (err %d)\n", err);
+
+	err = bt_mesh_provision(test_net_key, 0, 0, 0, 0x0001, dev_key);
+	ASSERT_OK(err, "Provisioning failed (err %d)", err);
+
+	for (int i = 0; i < PROV_REPROV_COUNT; i++) {
+		LOG_INF("Prov'ner prov loop #%d, waiting for prov ...\n", i);
+		ASSERT_OK(k_sem_take(&prov_sem, K_SECONDS(20)),
+			"Provisioner reprovision #%d fail", i);
+
+		err = bt_mesh_cfg_mod_sub_add(0, prov_addr - 1, 0x0000, 0xc000, BT_MESH_MODEL_ID_HEALTH_SRV, NULL);
+		ASSERT_OK(err, "Sub add failed (err %d)", err);
+		struct bt_mesh_cfg_mod_pub healthpub = {
+			.addr = 0xc001,
+			.app_idx = 0,
+			.cred_flag = false,
+			.ttl = 10,
+			.period = BT_MESH_PUB_PERIOD_10SEC(1),
+			.transmit = BT_MESH_TRANSMIT(3, 100)
+		};
+		err = bt_mesh_cfg_mod_pub_set(0, prov_addr - 1, 0x0000, BT_MESH_MODEL_ID_HEALTH_SRV, &healthpub, NULL);
+		ASSERT_OK(err, "Pub set failed (err %d)", err);
```

Comment:
Perhaps we should check that these are wiped after the reset before setting them again?

Reply:
... otherwise, the test wouldn't know for sure it has actually been reset internally. These APIs calls (in fact all configuration-related APIs) will be successful anyway if you provide the same inputs next time.